From: Sergio Durigan Junior <sergiodj@sergiodj.net>
Date: Thu, 22 Sep 2022 21:16:36 -0400
Subject: Initialize UID/GID when using popt in "main" functions

When using popt to parse the "--uid" and "--gid" parameters that can
be passed to providers/responders, we have to make sure to initialize
the corresponding UID and GID variables with zero otherwise they can
end up with garbage when no "--uid" nor "--gid" parameters are
provided.

Issue: #6347

Bugs: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1001377
      https://bugs.launchpad.net/ubuntu/+source/sssd/+bug/1989356

Signed-off-by: Sergio Durigan Junior <sergiodj@sergiodj.net>

Reviewed-by: Justin Stephenson <jstephen@redhat.com>
Reviewed-by: Sumit Bose <sbose@redhat.com>
(cherry picked from commit d91a814c3a1a32217ee1ab2d3569e3090648a418)

Origin: upstream, https://github.com/SSSD/sssd/commit/85c60e525e8e7d4c756027855b88d90fae389e91
Bug: https://github.com/SSSD/sssd/issues/6347
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/sssd/+bug/1989356
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1001377
---
 src/providers/data_provider_be.c  | 4 ++--
 src/providers/proxy/proxy_child.c | 4 ++--
 src/responder/autofs/autofssrv.c  | 4 ++--
 src/responder/ifp/ifpsrv.c        | 4 ++--
 src/responder/kcm/kcm.c           | 4 ++--
 src/responder/nss/nsssrv.c        | 4 ++--
 src/responder/pac/pacsrv.c        | 4 ++--
 src/responder/pam/pamsrv.c        | 4 ++--
 src/responder/ssh/sshsrv.c        | 4 ++--
 src/responder/sudo/sudosrv.c      | 4 ++--
 10 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/src/providers/data_provider_be.c b/src/providers/data_provider_be.c
index 5ceff04..d7f1e75 100644
--- a/src/providers/data_provider_be.c
+++ b/src/providers/data_provider_be.c
@@ -720,8 +720,8 @@ int main(int argc, const char *argv[])
     struct main_context *main_ctx;
     char *confdb_path;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/providers/proxy/proxy_child.c b/src/providers/proxy/proxy_child.c
index 54f1ea2..d23d04f 100644
--- a/src/providers/proxy/proxy_child.c
+++ b/src/providers/proxy/proxy_child.c
@@ -483,8 +483,8 @@ int main(int argc, const char *argv[])
     long id = 0;
     long chain_id;
     char *pam_target = NULL;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/autofs/autofssrv.c b/src/responder/autofs/autofssrv.c
index a14ab2c..afa94f5 100644
--- a/src/responder/autofs/autofssrv.c
+++ b/src/responder/autofs/autofssrv.c
@@ -179,8 +179,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/ifp/ifpsrv.c b/src/responder/ifp/ifpsrv.c
index d098f56..56311c8 100644
--- a/src/responder/ifp/ifpsrv.c
+++ b/src/responder/ifp/ifpsrv.c
@@ -305,8 +305,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/kcm/kcm.c b/src/responder/kcm/kcm.c
index 7ff3d02..a246235 100644
--- a/src/responder/kcm/kcm.c
+++ b/src/responder/kcm/kcm.c
@@ -324,8 +324,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/nss/nsssrv.c b/src/responder/nss/nsssrv.c
index f2ab724..1a84ee6 100644
--- a/src/responder/nss/nsssrv.c
+++ b/src/responder/nss/nsssrv.c
@@ -656,8 +656,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/pac/pacsrv.c b/src/responder/pac/pacsrv.c
index 518933f..960ccdb 100644
--- a/src/responder/pac/pacsrv.c
+++ b/src/responder/pac/pacsrv.c
@@ -174,8 +174,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/pam/pamsrv.c b/src/responder/pam/pamsrv.c
index 8311205..844c3a0 100644
--- a/src/responder/pam/pamsrv.c
+++ b/src/responder/pam/pamsrv.c
@@ -423,8 +423,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
     int pipe_fd = -1;
     int priv_pipe_fd = -1;
 
diff --git a/src/responder/ssh/sshsrv.c b/src/responder/ssh/sshsrv.c
index 95835a7..14c4cc4 100644
--- a/src/responder/ssh/sshsrv.c
+++ b/src/responder/ssh/sshsrv.c
@@ -164,8 +164,8 @@ int main(int argc, const char *argv[])
     char *opt_logger = NULL;
     struct main_context *main_ctx;
     int ret;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
diff --git a/src/responder/sudo/sudosrv.c b/src/responder/sudo/sudosrv.c
index 5fb9bfc..c5447d0 100644
--- a/src/responder/sudo/sudosrv.c
+++ b/src/responder/sudo/sudosrv.c
@@ -141,8 +141,8 @@ int main(int argc, const char *argv[])
     struct main_context *main_ctx;
     int ret;
     int pipe_fd = -1;
-    uid_t uid;
-    gid_t gid;
+    uid_t uid = 0;
+    gid_t gid = 0;
 
     struct poptOption long_options[] = {
         POPT_AUTOHELP
